pkg/agent/route/route_windows.go

```diff
@@ -174,6 +184,107 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
 	return nil
 }
 
+func (c *Client) AddServiceRoutes(svcIP net.IP, gwIP net.IP) error {
+	obj, found := c.hostRoutes.Load(svcIP.String())
+	svcIPNet := net.IPNet{IP: svcIP, Mask: net.CIDRMask(32, 32)}
+	metric := util.MetricHigh
+	route := &util.Route{
+		LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+		DestinationSubnet: &svcIPNet,
+		GatewayAddress:    gwIP,
+		RouteMetric:       metric,
+	}
+
+	if found {
+		existingRoute := obj.(*util.Route)
+		if existingRoute.GatewayAddress.Equal(route.GatewayAddress) && existingRoute.RouteMetric == route.RouteMetric {
+			klog.V(4).InfoS("Route already exists", "Service IP", route.DestinationSubnet.String(),
+				"Gateway", route.GatewayAddress.String(), "RouteMetric", route.RouteMetric)
+			return nil
+		}
+		// Remove the existing route if gateway or metric is not as expected.
+		if err := util.RemoveNetRoute(existingRoute); err != nil {
+			klog.ErrorS(err, "Failed to delete existing route entry",
+				"destination", existingRoute.DestinationSubnet.String(),
+				"gateway", existingRoute.GatewayAddress.String())
+			return err
+		}
+	}
+
+	// A virtual IP is used to avoid too many ARP responders in OVS flows.
+	// routes:
+	// 1. vRoute: Virtual IP -> antrea-gw0
+	// 2. svcRoute: SVC IP -> Virtual IP
+	vRoute := &util.Route{
```

Comment:
Is this route added every time when a new Service IP is added? I think only one time installation is OK, right?

Reply:
Yes. Updated.